### PR TITLE
[flutter_adaptive_scaffold] Support RTL

### PR DESCRIPTION
**Pull request by ahmednfwela**
This PR removes the hardcoded rtl directionality placed on top of `AdaptiveScaffold` for an unknown reason.

| RTL | LTR |
|-----|-----|
| image  |  image  |

## Changes in this PR

- Core: Removed this hardcoded `Directionality` widget https://github.com/flutter/packages/blob/050729760b251e315af01876cc7d7de5dcfba0e9/packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart#L491-L494
- Core: Added tests to ensure text direction is passed correctly
- Core: Modified `example/adaptive_scaffold_demo.dart` to demonstrate support for RTL
- Side: There were some tests that were getting skipped, I have enabled them since their related issues landed on stable.
- Side: Stopped ignoring `prefer_const_constructors` since it landed in stable as well.

I made the commit messages as descriptive as possible so that it's easier to review relevant changes.

## Related issues
- Supersedes #3602 
- Fixes https://github.com/flutter/flutter/issues/119661

Question for reviewers: should the next version be 0.1.5 or 0.2.0 ?

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks for fixing that!  We definitely shouldn't be hard coding a directionality.

**Comment by ahmednfwela** [comment]
@gspencergoog what should the next version be?

**Comment by gspencergoog** [comment]
> what should the next version be?
I'd say 0.1.5

File: packages/flutter_adaptive_scaffold/example/lib/main.dart [hunk 1]
```diff
@@ -52,6 +52,9 @@ class _MyHomePageState extends State<MyHomePage>
   // the navigation elements.
   ValueNotifier<bool?> showGridView = ValueNotifier<bool?>(false);
 
+  // Override the application's directionality
```
Comment (justinmc) [inline packages/flutter_adaptive_scaffold/example/lib/main.dart:55]: Nit: Period at the end.

**Review by justinmc — APPROVED** [review]
LGTM :+1: 

@gspencergoog It looks like there were some formatting changes in this PR, just making sure you are ok with that.

**Comment by justinmc** [comment]
Got it, thanks for the info!

**Comment by ahmednfwela** [comment]
@gspencergoog since the new version relies on dart 3/flutter 3.10, should this have been a minor version upgrade instead of a patch ? 
image

**Comment by gspencergoog** [comment]
No, it's not adding functionality, it's a bug fix. If we consider the updated dependency a breaking change, then it would be a major version update, but since this is still a 0. release, we won't bump it to 1.0 yet.

**Pull request merged**